File: src/com/gmail/marcosav2010/communicator/module/fth/packet/PacketFileRequest.java
package com.gmail.marcosav2010.communicator.module.fth.packet;

import java.io.IOException;

import com.gmail.marcosav2010.communicator.module.fth.FileSendInfo;
import com.gmail.marcosav2010.communicator.packet.Packet;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder;
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder;

import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
public class PacketFileRequest extends Packet {

	@Getter
	private String name;
	@Getter
	private int size;
	@Getter
	private int blocks;
	@Getter
	private int fileID;

	public PacketFileRequest(FileSendInfo info) {
		this.name = info.getPath().getFileName().toString();
		this.size = info.getSize();
		this.blocks = info.getBlocks();
		this.fileID = info.getFileID();
	}

	public boolean isSingle() {
		return blocks == 1;
	}

	@Override
	protected void encodeContent(PacketEncoder out) throws IOException {
		out.write(name);
		out.write(size);
		out.write(blocks);
		out.write(fileID);
	}

	@Override
	protected void decodeContent(PacketDecoder in) throws IOException {
		name = in.readString();
		size = in.readInt();
		blocks = in.readInt();
		fileID = in.readInt();
	}
}

File: modules/file-transfer/src/main/java/com/gmail/marcosav2010/fth/packet/PacketFileRequest.kt
package com.gmail.marcosav2010.fth.packet

import com.gmail.marcosav2010.communicator.packet.Packet
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketDecoder
import com.gmail.marcosav2010.communicator.packet.wrapper.PacketEncoder
import com.gmail.marcosav2010.fth.FileSendInfo
import kotlin.properties.Delegates

class PacketFileRequest() : Packet() {

    var name by Delegates.notNull<String>()
        private set

    var size by Delegates.notNull<Int>()
        private set

    var blocks by Delegates.notNull<Int>()
        private set

    var fileID by Delegates.notNull<Int>()
        private set

    val isSingle: Boolean
        get() = blocks == 1

    constructor(info: FileSendInfo) : this() {
        name = info.path.fileName.toString()
        size = info.size
        blocks = info.blocks
        fileID = info.fileID
    }

    override fun encodeContent(out: PacketEncoder) {
        out.write(name)
        out.write(size)
        out.write(blocks)
        out.write(fileID)
    }

    override fun decodeContent(`in`: PacketDecoder) {
        name = `in`.readString()
        size = `in`.readInt()
        blocks = `in`.readInt()
        fileID = `in`.readInt()
    }
}